icu4c/source/test/intltest/units_test.cpp

```diff
@@ -708,10 +708,22 @@ void UnitsTest::testComplexUnitsConverterSorting() {
 
     for (const auto &testCase : testCases) {
         MeasureUnitImpl inputImpl = MeasureUnitImpl::forIdentifier(testCase.input, status);
+        if (U_FAILURE(status)) {
+          return;
+        }
         MeasureUnitImpl outputImpl = MeasureUnitImpl::forIdentifier(testCase.output, status);
+        if (U_FAILURE(status)) {
+          return;
+        }
         ComplexUnitsConverter converter(inputImpl, outputImpl, conversionRates, status);
+        if (U_FAILURE(status)) {
+          return;
+        }
 
         auto actual = converter.convert(testCase.inputValue, nullptr, status);
+        if (U_FAILURE(status)) {
+          return;
+        }
```

Comment:
```suggestion
        if (status.errIfFailureAndReset()) {
            continue;
        }
        if (actual.length() < testCase.expectedCount) {
            errln("converter.convert(...) returned too few Measures");
            continue;
        }
```

Reply:
+1

@gnrunge , you can also add to the message the actual length and the expected length. (optional)

Reply:
Done.